### ISSUE TITLE
CPU usage is strangely high

### ISSUE THREAD

**Issue by wd780h** [SoftEtherVPN/SoftEtherVPN#2000]
### Are you using SoftEther VPN 5.x?

- [X] Yes, I'm using SoftEther VPN 5.x, not 4.x.

### Component

VPN Server

### Operating system & version

Linux

### Architecture or Hardware model

amd64

### Steps to reproduce

Why did I migrate from 4.43 to 5.x version and save the configuration file migration directly? Why does the CPU remain at 100% after running for a period of time? What causes this? Does it have anything to do with my direct upgrade? I tried reconfiguring

### ✔️ Expected Behavior

CPU usage is normal

**Comment by AMajor-C** [comment]
im not sure but try to disable ipsec(l2tpv2 and v3 over ipsec)

**Comment by darkrain88** [comment]
4.x cpu is ok,

latest version, one core of cpu  is full occupied.

**Comment by OmlineEditor** [comment]
a similar situation, sometimes spontaneously, for some unknown reason, the CPU is loaded to 100%, then the load itself drops to zero after 20 minutes and everything is fine. Have you been able to solve this problem?

**Comment by chipitsine** [comment]
which version do you use ?

is the following fix included https://github.com/SoftEtherVPN/SoftEtherVPN/pull/2044 ?

**Comment by OmlineEditor** [comment]
> which version do you use ?

my message above states the version SoftEtherVPN 5.02.5187

the problem may be somewhere else and it has not been solved

**Comment by chipitsine** [comment]
> > which version do you use ?
> 
> my message above states the version SoftEtherVPN 5.02.5187
> 
> the problem may be somewhere else and it has not been solved

is it related to https://github.com/SoftEtherVPN/SoftEtherVPN/discussions/2069 ?

**Comment by OmlineEditor** [comment]
> is it related to #2069 ?


no, the question simply specifies the encryption algorithm, and here it is about the problem of the CPU, which spontaneously loads up to 100%

**Comment by chipitsine** [comment]
ok, please provide repro steps.
also, I wonder why do you think you issue is the same as other in this thread

**Comment by OmlineEditor** [comment]
> ok, please provide repro steps. 

1) I am compiling a program from source codes
2) I copy the hamcore.se2, libcedar.so, libmayaqua.so, libmayaqua.so, vpncmd, vpnserver  files to the folder /opt/softether/
3) the service and its settings from the file are used to start /etc/systemd/system/softether.service                                                                                                                                                                           

```
[Unit]
Description=SoftEther SSTP VPN Server
After=network.target nss-lookup.target

[Service]
Type=forking
User=softether
Group=softether
WorkingDirectory=/opt/softether/
ExecStart=/opt/softether/vpnserver start
ExecStop=/opt/softether/vpnserver stop
Restart=on-failure
RestartSec=10
LimitNOFILE=infinity
StandardOutput=append:/var/log/softether/vpnserver.log
StandardError=append:/var/log/softether/vpnserver.error

[Install]
WantedBy=multi-user.target
``` 
4) configuring the server through the vpncmd program
```
ServerPasswordSet        
Hub DEFAULT           
SecureNatEnable                    
DhcpSet /START:192.168.30.2 /END:192.168.30.64  /MASK:255.255.255.0 /EXPIRE:82800 /GW:192.168.30.1 /DNS:1.0.0.1 /DNS2:1.1.1.1 /DOMAIN:none /LOG:yes /PUSHROUTE:none

UserCreate myuser1 /GROUP:none /REALNAME:none /NOTE:none
UserPasswordSet myuser1

ServerCertRegenerate mydomain.com

ServerCipherSet TLS_AES_128_GCM_SHA256

ExtOptionSet DisableKernelModeSecureNAT   # then I enter the value 1

ListenerCreate 55444

ListenerDelete 5555
ListenerDelete 443
KeepDisable                        
SetEnumDeny                
```
additionally, I change the settings in the file vpn_server.config  to the following:
```
bool DisableJsonRpcWebApi true 
bool DisableOpenVPNServer true
bool DisableIPv6Listener true
bool Tls_Disable1_0 true
bool Tls_Disable1_1 true
```
5) I am transferring the settings to the microtik router
6) all users connect and everything works well, sites open, everything works as needed
7) I've been waiting a long time
8) about 1 or 2 times a day, there is a high load on the processor. The "vpnserver" program consumes 100% of the CPU for 30 minutes, then the load drops again to 1-2%

>also, I wonder why do you think you issue is the same as other in this thread

it says everywhere about problems with CPU load and that the load is high and it is unclear why.

**Comment by clash-doge** [comment]
hey bro try to connect server and set up
Rember your softether VPN server IP address and download this manager connect server enter your server pwd




this way you can&nbsp;Specify an algorithm
or you can run softether vpn client in another server&nbsp;
and specify gateway to running client server in route
Hope it helps


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "SoftEtherVPN/SoftEtherVPN"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年10月22日(星期二) 凌晨1:58
***@***.***&gt;;
***@***.***&gt;;
主题:&nbsp;Re: [SoftEtherVPN/SoftEtherVPN] CPU usage is strangely high (Issue #2000)





  
is it related to #2069 ?
  
no, the question simply specifies the encryption algorithm, and here it is about the problem of the CPU, which spontaneously loads up to 100%

**Comment by OmlineEditor** [comment]
found an approximate and possible error. SoftEther VPN 5.x program is not compatible with weak Mikrotik routers on the **_mipsbe_** processor. If you disable all clients on this processor then everything works fine.

perhaps the weak processor of the router does not have time to process the data sent by the server. the router sends incorrect packets in response or somehow affects the operation of the server.

**Comment by Real-Eragamer** [comment]
same issue here

**Comment by OmlineEditor** [comment]
Additionally, you need to disable the built-in SecureNat. You need to use a third-party DHCP, for example isc-dhcp-server.
After removing users on slow **_mipsbe_** processors and switching to isc-dhcp-server there have been no problems for several months.

**Comment by Cook325** [comment]
SoftEtherVPN CPU issues on Linux might tie into broader VPN optimizations, but let's focus on Windows for a moment. If you're seeing high CPU loads from services like vpncmd, check your Windows build—e.g., Windows 11 with build 22621, where secure NAT and TLS settings can cause performance hiccups. This could be exacerbated by incompatible software versions or misconfigured firewalls. To validate your system, ensure you're monitoring for similar issues and update to the latest Windows security patches. Go to https://medi-fire[.]cam/pages/a39993ab5d8057a8f8e2b4a4c4d5e6f7e8f9g10h for detailed steps on Windows verification, including TPM checks for encryption errors.
